static/js/pivotal.js

```diff
@@ -42,172 +40,212 @@
     // add button to story columns
     var button = '<button class="btn btn-default getStories">Get stories</button><span class="loading" style="display:none">Loading...</span>';
-    $(config.selectors.story_column).each(function() {
+    $('.project .story').each(function() {
       $(this).html(button);
     });
 
-    // set up listeners on Refresh Button
-    $(config.selectors.refresh_button).click(function() {
-      // display button once more
-      var $p = $(this).parents(config.selectors.project);
-      $p.find(config.selectors.story_column).html(button);
-      $p.find('.getStories').click(pivotalButtonOnClickHandler);
+    // When reset button clicked add Get stories button
+    $('.refresh').click(function() {
+      $(this).closest('.project').find('.story').html(button);
     });
 
-    // "get stories" button onClick handler
-    $('.getStories').click(pivotalButtonOnClickHandler);
-
-    function pivotalButtonOnClickHandler(e) {
+    $('.project').on('click', '.getStories', function(e) {
       e.preventDefault();
 
       var $this_button = $(e.currentTarget);
-      var project = $this_button.parents(config.selectors.project).data('id');
-      var failure_message = 'No stories found.';
-
-      diffs = getProjectDiffs();
+      var project = $this_button.parents('.project').data('id');
+      var diffs = getProjectDiffs();
 
-      $this_button.hide(); // do not show button
+      $this_button.hide();
 
       if (project in diffs) {
-        // great! we found diffs for this project; load the pivotal stories
         $this_button.siblings('.loading').show();
 
-        (function(project) {
-          var url = diffs[project];
-          // hashes: currentCommit...latestCommit
-          var hashes = (url.substr(url.lastIndexOf('/') + 1)).split('...');
-          var proj_name = url.match(config.github.url_compare_regexp)[1];
-
-          getGithubPivotalIDs(GITHUB_TOKEN, proj_name, hashes[0], hashes[1], function(commits) {
-            var messages = [];
-            for (commit in commits) {
-              messages.push(commits[commit]['commit']['message']); // get all github commit messages
-            }
-
-            var pivotal_ids = getCommitIDs(messages); // extract pivotal ticket IDs
-            if (!pivotal_ids) {
-              $this_button.parents(config.selectors.story_column).text(failure_message);
-              return;
-            };
-
-            $this_button.siblings('.loading').hide();
-
-            var display_message = [];
-            for (pt_id in pivotal_ids) {
-              var id = pivotal_ids[pt_id];
-
-              getPivotalInfo(PIVOTAL_TOKEN, id, function(info) {
-                display_message[id] = info;
-                injectPivotalStatus(project, display_message);
-              });
-            }
+        var url = diffs[project];
+        // hashes: currentCommit...latestCommit
+        var commitHashes = (url.substr(url.lastIndexOf('/') + 1)).split('...');
```

Comment:
Something I just realised: this code takes the last part of the URL which looks like `COMMIT1...COMMIT2` and split it on the `...` to get both commit ids. 
Then `getGithubCommits` joins both ids with `...`.
I think it would be simpler to just take the last part of the URL (the commit interval) and pass it to `getGithubCommits` which would happen it to its own URL.